tests/api_resources/test_topup.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import absolute_import, division, print_function
+
+import stripe
+from tests.helper import StripeTestCase
+
+
+TEST_RESOURCE_ID = 'tu_123'
+
+
+class TopupTest(StripeTestCase):
+    def test_is_listable(self):
+        resources = stripe.Topup.list()
+        self.assert_requested(
+            'get',
+            '/v1/topups'
+        )
+        self.assertIsInstance(resources.data, list)
+        self.assertIsInstance(resources.data[0], stripe.Topup)
+
+    def test_is_retrievable(self):
+        resource = stripe.Topup.retrieve(TEST_RESOURCE_ID)
+        self.assert_requested(
+            'get',
+            '/v1/topups/%s' % TEST_RESOURCE_ID
+        )
+        self.assertIsInstance(resource, stripe.Topup)
+
+    def test_is_creatable(self):
+        resource = stripe.Topup.create(
+            amount=100,
+            currency='usd',
+            source='src_123',
+            description='description',
+            statement_descriptor='statement descriptor'
+        )
+        self.assert_requested(
+            'post',
+            '/v1/topups'
+        )
+        self.assertIsInstance(resource, stripe.Topup)
+
+    def test_is_saveable(self):
+        resource = stripe.Topup.retrieve(TEST_RESOURCE_ID)
+        resource.metadata['key'] = 'value'
+        resource.save()
+        self.assert_requested(
+            'post',
+            '/v1/topups/%s' % resource.id
+        )
+
+    def test_is_modifiable(self):
+        resource = stripe.Topup.modify(
+            TEST_RESOURCE_ID,
+            metadata={'key': 'value'}
+        )
+        self.assert_requested(
+            'post',
+            '/v1/topups/%s' % TEST_RESOURCE_ID
+        )
+        self.assertIsInstance(resource, stripe.Topup)
```

Comment:
Worth checking the `resource` has its `metadata` set to `{'key': 'value'}`?

Reply:
I tried this, but it looks like stripe-mock doesn't actually mutate received data correctly. I believe it just validates what it received and returns stock data.

Reply:
👍

Reply:
Yes that's correct Jamu. We want to fix this at some point, but for now it's not working checking any of the data that you get back.